doc/architecture/overview.md

```diff
@@ -0,0 +1,205 @@
+
+Donations can also dropped off at **Donation Sites**. These are locations that have been designated as places where people can drop off donations.
+
+Donations that don't fit into either of those types are classified as "Miscellaneous Donations."
+
+Both Donation Sites and Diaper Drives have to be created in the application beforehand, but can be used repeatedly after that.
+
+### Inventory Management
+
+All inventory is physically stored at **Storage Locations**. This is one area where the application is particularly useful, as some storage locations accrue large quantities of inventory over time. Each storage location can contain any number of **Items** or types of items, and if a diaper bank has multiple storage locations, it is possible for the same item type to exist at multiple locations.
+
+Sometimes, the diaper banks will need to make a correction, this application calls them **Adjustment**s.
+
+When a diaper bank wants to move inventory from one storage location to another, they do so with a **Transfer**.
+
+### Partner Distribution
+
+When inventory leaves a diaper bank, it does so via a **Distribution**. These are either created implicitly from inbound Partner **Requests**, or created explicitly via the menu interface. When they are created explicitly, they pull from a single designated Storage Location, and are built in a similar fashion to Donations, Adjustments, Transfers, etc -- items are added and quantities are specified.
+
+Distributions can be exported as PDFs, which Banks can use as printable manifests for the packages sent to the Community Partner.
+
+Community Partners then dispense the items they receive to the clients / families.
+
+# Application Architecture
+
+This section is a more detailed, and more technical, explanation of how the application works, internally.
+
+## Multi-Tenancy
+
+![Multi Tenancy](./multi-tenancy.svg)
+<details>
+<summary>(Diagram Code - Multi Tenancy)</summary>
+
+```plantuml:multi-tenancy
+skinparam componentStyle rectangle
+skinparam actorStyle awesome
+
+'skinparam linetype ortho
+
+actor "User A\n(Admin)" as user_a
+actor "User B" as user_b
+actor "User C" as user_c
+actor "User D" as user_d
+
+cloud "humanessentials.app" {
+
+user_a --> [Bank A]
+user_b --> [Bank A]
+
+[Bank A] --> [Partner A]
+[Bank A] --> [Partner B]
+[Bank A] --> [Partner C]
+
+user_b --> [Bank B]
+[Bank B] --> [Partner C]
+
+user_c -u-> [Partner A]
+user_d --> [Bank A]
+user_d --> [Partner A]
+}
+```
+</details>
+
+This application is multi-tenant -- that is, each Diaper Bank (used interchangeably with **Organization**) has its own templated "section" of the application, and may act in that section without concern that its changes will affect other organizations (and vice versa).
+
+When a user is signed-in, they are automatically "isolated" to their organizational space, as indicated by the URL (which features a short-code of their organization in it).
```

Comment:
With [this PR](https://github.com/rubyforgood/human-essentials/pull/3386) this won't really be true any more - the organization ID in the URL isn't relevant, it's the current role of the user. I'm not even sure that referring to the app as "multi-tenant" is true; usually a multi-tenanted app refers to the database structure (where each client has their own table or set of tables).

Reply:
I thought multi-tenant meant multiple-organizations on a single-db instead of each org having their own db. But really I copy/pasted this from a previous thing, so I'm simplifying :)